src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -151,7 +151,15 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
             }
             else
             {
-                operand = this.bindMethod(queryToken) as CollectionNode;
+                var node = this.bindMethod(queryToken);
+                if (node is SingleValueOpenPropertyAccessNode openNode)

```

Comment:
why doesn't the 'bindMethod' return 'CollectionOpenPropertyAccessNode' directly?